packages/create-webstone-app/src/index.ts

```diff
@@ -1,6 +1,155 @@
-import { displayWelcome, displayNextSteps } from "./helpers";
-import { tasks } from "./tasks/index";
+import { CreateWebstoneOptions, WebstoneAppType } from "../types";
+import fs from "fs-extra";
+import path from "node:path";
+import { PackageJson } from "type-fest";
+import { appPackageJson, pluginPackageJson } from "./package";
+import merge from "ts-deepmerge";
+import { create } from "create-svelte";
 
-displayWelcome();
-const context = await tasks.run();
-displayNextSteps(context);
+export async function createWebstone(
+  cwd: string,
+  options: CreateWebstoneOptions,
+) {
+  const { type, extendCLI } = options;
+  const appName = getAppName(cwd, type);
+  createBaseApp(cwd, { type, appName });
+  updatePackageJSON(cwd, { type, extendCLI });
+  if (type === "plugin" && extendCLI) {
+    copyBuildScript(cwd);
+    copyCLIExtension(cwd);
+  }
+}
+
+function getRawAppName(cwd: string) {
+  let appName: string = cwd;
+  if (cwd === ".") {
+    appName = process.cwd().split("/").pop() || "webstone-project";
+  }
+  return appName;
+}
+
+function getAppName(cwd: string, type: WebstoneAppType) {
+  let appName = getRawAppName(cwd);
+  if (type === "plugin") {
+    appName = `webstone-plugin-${appName}`;
+    return toValidPackageName(appName);
+  }
+  return toValidPackageName(appName);
+}
+
+function toValidPackageName(name: string) {
+  return name
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, "-")
+    .replace(/^[._]/, "")
+    .replace(/[^a-z0-9~.-]+/g, "-");
+}
+
+function createBaseApp(
+  cwd: string,
+  options: { type: WebstoneAppType; appName: string },
+) {
+  const { appName, type } = options;
+  create(cwd, {
+    name: appName,
+    template: type === "app" ? "skeleton" : "skeletonlib",
+    types: "typescript",
+    eslint: true,
+    prettier: true,
+    playwright: true,
+    vitest: true,
+  });
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function sortKeys(obj: any) {
+  if (typeof obj !== "object" || obj === null)
+    throw new Error("Invalid object");
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const sortedObj: Record<string, any> = {};
+  Object.keys(obj)
+    .sort()
+    .forEach((key) => {
+      sortedObj[key] = obj[key];
+    });
+  return sortedObj;
+}
+
+function updatePackageJSON(
+  cwd: string,
+  options: { type: WebstoneAppType; extendCLI: boolean },
+) {
+  const { type, extendCLI } = options;
+  const pkg: PackageJson = fs.readJSONSync(path.resolve(`${cwd}/package.json`));
+  let newPkg: PackageJson = pkg;
+  if (type === "app") {
+    newPkg = deepMergeWithSortedKeys(pkg, appPackageJson);
+  }
+  if (type === "plugin" && extendCLI) {
+    newPkg = deepMergeWithSortedKeys(pkg, pluginPackageJson);
+  }
+
+  fs.writeJsonSync(path.resolve(`${cwd}/package.json`), newPkg, {
+    encoding: "utf-8",
+    spaces: "\t",
```

Comment:
Nice, I'm on the tabs train as well. Took me two decades and one Rich Harris, but I finally realized it's the better option.